src/zope/app/onlinehelp/__init__.py

```diff
@@ -116,18 +138,13 @@ def getTopicFor(obj, view=None):
     True
 
     """
-    topic = None
     for interface in providedBy(obj):
-        for t in getUtilitiesFor(IOnlineHelpTopic):
-            if t[1].interface==interface and t[1].view==view:
-                topic = t[1]
-                break
-
-    return topic
+        for _name, topic in getUtilitiesFor(IOnlineHelpTopic):
+            if topic.interface == interface and topic.view == view:
+                return topic
```

Comment:
This is an incidental fix I noticed. This method is specifically documented as "Returns the first match", but in reality it was returning the first match for the *last* interface the object provides (i.e., the least derived), if there was one.

There is not a specific test case for this.